source/includes/fundamentals/code-snippets/Monitoring.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    private void monitorCommandEvent() {
+        // start monitor-command-example
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .addCommandListener(new CommandCounter())
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run some commands to test the timer
+        collection.find().first();
+        collection.find().first();
+        mongoClient.close();
+        // end monitor-command-example
+    }
+
+    private void monitorClusterEvent() {
+        // start monitor-cluster-example
+        IsWriteable clusterListener = new IsWriteable();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToClusterSettings(builder ->
+                                builder.addClusterListener(clusterListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run a command to trigger a ClusterDescriptionChangedEvent event
+        collection.find().first();
+        // end monitor-cluster-example
+        mongoClient.close();
+    }
+
+    private void monitorConnectionPoolEvent() {
+        // start monitor-cp-example
+        ConnectionPoolLibrarian cpListener = new ConnectionPoolLibrarian();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToConnectionPoolSettings(builder ->
+                                builder.addConnectionPoolListener(cpListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run a command to trigger connection pool events
+        collection.find().first();
+        // end monitor-cp-example
+        /* We do not close this connection in order to prevent the driver from requesting two connections, giving
+        the example unintuitive output. Uncomment the following line of code and see how it effects this example*/
+        //mongoClient.close();
+    }
+
+}
+
+// start command-listener-impl
+class CommandCounter implements CommandListener {
+
+    private Map<String, Integer> commands = new HashMap<String, Integer>();
+
+    @Override
+    public void commandSucceeded(final CommandSucceededEvent event) {
```

Comment:
`public synchronized void`

Reply:
Thanks for the note. Didn't realize that Java `HashMap` is not thread safe (and was not considering this class being used in a multithreaded app). Fix: https://github.com/mongodb/docs-java/pull/88/commits/d45b8ba06cb1687ce13a742cc722cefa54a81f99

Going forward, I will try to think about structuring examples for multithreaded environments.